motor/providers/k8s/manifest_parser.go

```diff
@@ -438,6 +439,47 @@ func (t *manifestParser) DaemonSets(namespace v1.Namespace) ([]*appsv1.DaemonSet
 	return sliceToPtrSlice(daemonsets), nil
 }
 
+func (t *manifestParser) Ingress(namespace, name string) (*networkingv1.Ingress, error) {
+	result, err := t.Resources("ingresses.v1.networking.k8s.io", name, namespace)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(result.Resources) > 1 {
+		return nil, errors.New("multiple ingresses found")
```

Comment:
I know this is not part of this PR but perhaps we want to improve this error message (also for the other resources). If we show this message to a user, it will be very confusing. Yes, there are multiple ingresses found but that is not the problem. The problem is they have the same name and namespace.,

Reply:
This is for a manfiest, so technically there's nothing stopping you from having the same resource twice in a single file. To make it more complicated, now that we have recursive directory scanning you can end up in this situation even without having the same resource in a single file

Reply:
Ok, makes sense to me. But this could also be valid. I opened an issue for that: https://github.com/mondoohq/cnquery/issues/616

But, I think, this needs some more discussion.